### ISSUE TITLE
[Feature Request]: Add general thumbnail support for file types with thumbnails in zip file (i.e. Libre-/OpenOffice)

### ISSUE THREAD

**Issue by Heiholf** [TagStudioDev/TagStudio#379]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

LibreOffice files only show up as icons 
![image](https://github.com/user-attachments/assets/670a38f6-dda6-4220-a852-9dc07ab453e9)

### Solution

As described in #363 the thumbnails are part of the file. The file is stored as a zip and the thumbnail is stored under "Thumbnails/thumbnail.png". 

It would be quite easy to do the same as #366 for these files. However, as there are probably more files like these it might be useful to bundle the functionality for file types with this structure.

I don't know how easy it is right now or if it is more useful to wait until [thumbnails](/TagStudioDev/TagStudio/tree/thumbnails) is merged into [main](/TagStudioDev/TagStudio/tree/main).

Some file types which have this kind of pattern are (some where added afterwards):
- [x] .mscz (MuseScore)
- [ ] .odf .odp .ods .odt .odg (LibreOffice/OpenOffice)
- [ ] .ora (OpenRaster [Qronikarz](https://github.com/TagStudioDev/TagStudio/issues/379#issuecomment-2307582992))
- [ ] .kra (Krita [blipdrifter](https://github.com/TagStudioDev/TagStudio/issues/379#issuecomment-2336399524))

### Alternatives

Alternatively, one could handle all of these files individually which would create an unnecessary amount of repetition in the code.

**Comment by Qronikarz** [comment]
So .ora images work the same way. Good to know.

More info about .ora
https://github.com/orgs/linuxmint/discussions/562

tl;dr
Multi layered image format for image editors. Used for example in Pinta.

Would you mind adding it to the list?

**Comment by FreeCap23** [comment]
Hi, are .kra file types supported? I see the issue is closed, but there are no thumbnails for me on Krita projects.

**Comment by purpletennisball** [comment]
> Hi, are .kra file types supported? I see the issue is closed, but there are no thumbnails for me on Krita projects.

It seems that the PR that closed the issue (#545) was only for OpenDocument type files.

**Comment by FreeCap23** [comment]
Should this issue be re-opened or should a new one be created?

**Comment by CyanVoxel** [comment]
A new issue would be better - and it should be trivial to add .kra and .ora thumbnail support if they have a similar zip file structure, I'll just need to get my hands on some example files
